docs/ecosystem-roadmap.md

```diff
@@ -10,92 +10,108 @@ ms.prod: console
 
 ### Device
 
-The device is an intermediate message handling communications layer between two processes, the client and the server. On Windows, this is the console driver, while on other platforms, it is the TTY or PTY device. Other devices like files, pipes, and sockets may be used as this communication channel if the entire transaction is in plain text or contains [virtual terminal sequences](console-virtual-terminal-sequences.md), but not with [Windows Console APIs](console-functions.md).
+The device is an intermediate message-handling communications layer between two processes, the client and the server. On Windows, this is the console driver. On other platforms, it is the TTY or PTY device. Other devices like files, pipes, and sockets may be used as this communication channel if the entire transaction is in plain text or contains [virtual terminal sequences](console-virtual-terminal-sequences.md), but not with [Windows Console APIs](console-functions.md).
 
 ### Server
 
-The server interprets the requested API calls or messages from the client. On Windows in the classic operating mode, it also creates a user interface to present the output to the screen and collects input to send back in response messages to the client via the driver, like a terminal bundled in the same module. With the [pseudoconsole](pseudoconsoles.md) mode, it instead is only a translator to present this information in [virtual terminal sequences](console-virtual-terminal-sequences.md) to an attached Terminal.
+The server interprets the requested API calls or messages from the client. On Windows in the classic operating mode, the server also creates a user interface to present the output to the screen. The server additionally collects input to send back in response messages to the client, via the driver, like a terminal bundled in the same module. Using [pseudoconsole](pseudoconsoles.md) mode, it instead is only a translator to present this information in [virtual terminal sequences](console-virtual-terminal-sequences.md) to an attached terminal.
 
 ### Terminal
 
-The terminal is the final layer providing graphical display and interactivity services to the user. It is responsible for capturing input and encoding it as [virtual terminal sequences](console-virtual-terminal-sequences.md) to eventually reach the client's `STDIN` and decoding and presenting the plain text and _virtual terminal sequences_ it received back from the client's `STDOUT`.
+The terminal is the final layer providing graphical display and interactivity services to the user. It is responsible for capturing input and encoding it as [virtual terminal sequences](console-virtual-terminal-sequences.md). These virtual terminal sequences eventually reach the clients `STDIN`, decoding and presenting the plain text and _virtual terminal sequences_ it received back from the clients `STDOUT`.

```

Comment:
I don't really like where this went. I feel like you were trying to split the run on. But I'd rather the "capturing input and encoding" clause go with the `STDIN` one and then split off the `STDOUT` one as a standalone sentence if that was the goal.

Reply:
Yes - I've tried to break up some of the long sentences to make them more readable / digestable. How about this?

The terminal is the final layer providing graphical display and interactivity services to the user. It is responsible for capturing input and encoding it as [virtual terminal sequences](console-virtual-terminal-sequences.md), which eventually reach the clients `STDIN`. Once the virtual terminal sequences reach the 'STDIN`, they are decoded and present the plain text, along with the _virtual terminal sequences_ that it received back from the clients `STDOUT`.

Reply:
Ehhh, that sort of still mixes the role of who is doing the decoding of what. I'm really just talking about from the Terminal's perspective here so I'd revise one further as:

The terminal is the final layer providing graphical display and interactivity services to the user. It is responsible for capturing input and encoding it as [virtual terminal sequences](console-virtual-terminal-sequences.md), which eventually reach the client's `STDIN`. It will also receive and decode the *virtual terminal sequences* that it receives back from the client's `STDOUT` for presentation on the screen.